docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "To do this, you must:\n",
+    "- Provide `with_repo=True` when creating your function via `project.set_function(...)`\n",
+    "- Set project source via `project.set_source(source=...)`\n",
+    "\n",
+    "This instructs MLRun to load source code from the git repo/archive/file share associated with the project. There are two ways to load these additional files:\n",
+    "\n",
+    "#### Load Code from Container\n",
+    "The function is built once. *This is the preferred approach for production workloads*:\n",
+    "```python\n",
+    "project.set_source(source=\"git://github.com/mlrun/project-archive.git\")\n",
+    "\n",
+    "fn = project.set_function(\n",
+    "    name=\"myjob\", handler=\"job_func.job_handler\",\n",
+    "    image=\"mlrun/mlrun\", kind=\"job\", with_repo=True,\n",
+    ")\n",
+    "\n",
+    "project.build_function(fn)\n",
+    "```\n",
+    "\n",
+    "#### Load Code at Runtime\n",
+    "The function pulls the source code at runtime. *This is a simpler approach during development that allows for making code changes without re-building the image each time*:\n",
+    "\n",
+    "```python\n",
+    "archive_url = \"https://s3.us-east-1.wasabisys.com/iguazio/project-archive/project-archive.zip\"\n",
+    "project.set_source(source=archive_url, pull_at_runtime=True)\n",
+    "\n",
+    "fn = project.set_function(\n",
+    "    name=\"nuclio\", handler=\"nuclio_func:nuclio_handler\",\n",
+    "    image=\"mlrun/mlrun\", kind=\"nuclio\", with_repo=True,\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "969cc92e",
+   "metadata": {},
+   "source": [
+    "<a id=\"import-existing-function\"></a>\n",
+    "### Import or Use Existing Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c5adbe0",
+   "metadata": {},
+   "source": [
+    "If you already have an MLRun function that you would like to import, you can do so from multiple locations such as YAML, function marketplace, and MLRun DB."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "36390042",
+   "metadata": {},
+   "source": [
+    "#### YAML"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ed7f8c84",
+   "metadata": {},
+   "source": [
+    "MLRun functions can be exported to YAML files via `fn.export()`. These YAML files can then be imported via the following:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d5f5e26",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "fn = project.set_function(name=\"import\", func=\"function.yaml\")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94ffa4b2",
+   "metadata": {},
+   "source": [
+    "#### Function Marketplace"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a65c196",
+   "metadata": {},
+   "source": [
+    "Functions can also be imported from the [**MLRun Function Marketplace**](https://www.mlrun.org/marketplace) - simply import via the name of the function and the `hub://` prefix:\n",
```

Comment:
"Functions can also be imported from the [**MLRun Function Marketplace**](https://www.mlrun.org/marketplace). Simply import via the name of the function and the `hub://` prefix:\n",